Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def actual_url(url)
+    m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
```

Comment:
It slows the GNU formulae `fetch` process down slightly. That's the only problem I've seen though. I `fetch`ed several SF formulae, several known non-redirected formulae, bottles and source packages. What problem are you expecting to run into here?